src/core/vectortile/qgsmapboxglstyleconverter.cpp

```diff
@@ -1433,6 +1460,33 @@ bool QgsMapBoxGlStyleConverter::parseSymbolLayerAsRenderer( const QVariantMap &j
         markerDdProperties.setProperty( QgsSymbolLayer::PropertyWidth, QgsProperty::fromExpression( spriteSizeProperty ) );
       }
 
+      if ( jsonLayout.contains( QStringLiteral( "icon-size" ) ) )
+      {
+        const QVariant jsonIconSize = jsonLayout.value( QStringLiteral( "icon-size" ) );
+        double size;
+        switch ( jsonIconSize.type() )
+        {
+          case QVariant::Int:
+          case QVariant::Double:
+            size = jsonIconSize.toDouble() * context.pixelSizeConversionFactor();
```

Comment:
I don't think we should use the pixel ratio here -- see https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-symbol-icon-size the property is in percent of original sprite size